dissect/target/tools/diff.py

```diff
@@ -0,0 +1,961 @@
+                if match:
+                    continue
+            yield from self.walkdir(subdirectory.path, exclude, already_iterated)
+
+    def differentiate_plugin_outputs(
+        self, plugin_name: str, plugin_arg_parts: list[str], only_changed: bool = False
+    ) -> Iterator[Record]:
+        """Run a plugin on the source and destination targets and yield RecordUnchanged, RecordCreated and RecordDeleted
+        records. There is no equivalent for the FileModifiedRecord. For files and directories, we can use the path to
+        reliably track changes from one target to the next. There is no equivalent for plugin outputs, so we just assume
+        that all records are either deleted (only on src), created (only on dst) or unchanged (on both)."""
+        with ignore_fields_for_comparison(["_generated", "_source", "hostname", "domain"]):
+            src_records = set(get_plugin_output_records(plugin_name, plugin_arg_parts, self.src_target))
+            src_records_seen = set()
+
+            for dst_record in get_plugin_output_records(plugin_name, plugin_arg_parts, self.dst_target):
+                if dst_record in src_records:
+                    src_records_seen.add(dst_record)
+                    if not only_changed:
+                        yield RecordUnchangedRecord(
+                            src_target=self.src_target.path, dst_target=self.dst_target.path, record=dst_record
+                        )
+                else:
+                    yield RecordCreatedRecord(
+                        src_target=self.src_target.path, dst_target=self.dst_target.path, record=dst_record
+                    )
+            for record in src_records - src_records_seen:
+                yield RecordDeletedRecord(
+                    src_target=self.src_target.path, dst_target=self.dst_target.path, record=record
+                )
+
+
+class DifferentialCli(ExtendedCmd):
+    """CLI for browsing the differential between two or more targets."""
+
+    doc_header_prefix = "Target Diff\n" "==========\n"
+    doc_header_suffix = "\n\nDocumented commands (type help <topic>):"
+    doc_header_multiple_targets = "Use 'list', 'prev' and 'next' to list and select targets to differentiate between."
+
+    def __init__(self, *targets: tuple[Target], deep: bool = False, limit: int = FILE_LIMIT):
+        self.targets = targets
+        self.deep = deep
+        self.limit = limit
+
+        self.src_index = 0
+        self.dst_index = 0
+        self.comparison: TargetComparison = None
+
+        self.cwd = "/"
+        self.alt_separator = "/"
+
+        doc_header_middle = self.doc_header_multiple_targets if len(targets) > 2 else ""
+        self.doc_header = self.doc_header_prefix + doc_header_middle + self.doc_header_suffix
+
+        self._select_source_and_dest(0, 1)
+        if len(self.targets) > 2:
+            # Some help may be nice if you are diffing more than 2 targets at once
+            self.do_help(arg=None)
```

Comment:
I think this line should be moved after the `super().__init__()` otherwise you get an error:
`AttributeError: 'DifferentialCli' object has no attribute 'stdout'`